### PR TITLE
Add function for accounting of debt due to delegation

### LINKED ISSUE #238: Add function to track delegate fees to currency networks
Add function to set / update mapping `uniqueID => value` where value is the owed value from the point of view of the lowest ordered address.

See: https://github.com/trustlines-protocol/contracts/blob/master/contracts/CurrencyNetwork.sol#L1461

### PR DESCRIPTION
**Pull request by cducrest**
closes https://github.com/trustlines-protocol/contracts/issues/238

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
I don't understand how increaseDebt is going to be used. Please add a comment that explains it.

**Comment by cducrest** [comment]
> I don't understand how increaseDebt is going to be used. Please add a comment that explains it.

Done

**Comment by schmir-at-bb** [comment]
> > I don't understand how increaseDebt is going to be used. Please add a comment that explains it.
> 
> Done

I think the crucial information is that this is going to be called from the identity contract. That information is still missing.

**Review by ghost — APPROVED** [review]

**Review by berndbohmeier — CHANGES_REQUESTED** [review]
Please lets have a short discussion about this tomorrow.

File: contracts/CurrencyNetwork.sol [hunk 1]
```diff
@@ -521,6 +523,39 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
...
+        address debtor = msg.sender;
+        int72 oldDebt = debt[uniqueIdentifier(debtor, creditor)];
+        if (debtor < creditor) {
+            int72 newDebt = oldDebt + value;
+            assert(newDebt > oldDebt);
+            debt[uniqueIdentifier(debtor, creditor)] = newDebt;
+        } else {
+            int72 newDebt = oldDebt - value;
+            assert(newDebt < oldDebt);
+            debt[uniqueIdentifier(debtor, creditor)] -= value;
```
Comment (berndbohmeier) [inline contracts/CurrencyNetwork.sol]:
```suggestion
            debt[uniqueIdentifier(debtor, creditor)] = newDept;
```

**Review by berndbohmeier — APPROVED** [review]

**Comment by berndbohmeier** [comment]
I would not document how it will be used. But we should probably later document how the tracked debt can be redeemed.

**Comment by schmir-at-bb** [comment]
> I would not document how it will be used, as this is the wrong direction of thinking and creates an additional dependency. But we should probably later document how the tracked debt can be redeemed.

Well, the function is there and there's a reason for the existence of that function. Why would any sane person call a function called increaseDebt, when it's your own debt that's increasing? I think that's rather surprising and deserves some clarification.

**Comment by berndbohmeier** [comment]
Yeah, but this is just one usecase. It is in general there to be able to have a a debt claim for trustlines money. 
I feel like you could do the same argumentation for a transfer, why should a sane person do a transfer when then the other person receives the money.

**Comment by berndbohmeier** [comment]
We can document it as `Provides a way to pay someone if there is no trustlines path yet. The claim can be redeemed later if there exists a path then`

Would that work for you?

**Comment by schmir-at-bb** [comment]
>  I feel like you could do the same argumentation for a transfer, why should a sane person do a transfer when then the other person receives the money.

That's right. Yet, it's still confusing to me, even more so because I'm making up bogus arguments. Just let me not think about it by telling my why it's here and how it's being used.

> Provides a way to pay someone if there is no trustlines path yet. The claim can be redeemed later if there exists a path then

That's ok, though I still do think it wouldn't hurt to say it was introduced to pay delegates via calls from the identity contract.

**Pull request closed without merging**